### ISSUE TITLE
Residual plot is inverted

### ISSUE THREAD

**Issue by abonawas** [lmfit/lmfit-py#457]
Example:
```
import numpy as np
import matplotlib.pyplot as plt
from lmfit import Model

f = lambda x: x - x +12
x = np.arange(-5, 5)
y = x**2

linear_model = Model(f)
params = linear_model.make_params()
result = linear_model.fit(y, x=x)

result.plot()
plt.show()
```

I checked the Model class, the method plot_residuals should return:
`ax.plot(x_array, self.data - self.eval() , datafmt,
                    label='residuals', **data_kws)`
Instead of:
`ax.plot(x_array, self.data - self.eval() , datafmt,
                    label='residuals', **data_kws)`

Thank you,
Abonawas.

**Comment by newville** [comment]
@abonawas We absolutely assert that you have read and understood Contributing.md before raising an Issue.  And yet you still decided (deliberately and incorrectly) to raise an Issue instead of asking a question.  In fact, the `plot_residuals()` method does

   ax.plot(x_array, self.eval() - self.data, datafmt,  label='residuals', **data_kws)

so your Issue is Just Plain Wrong.     

Are you trying to assert that "fit-data" is wrong and it should be "data-fit"?  Who knows? 

Use the mailing list for questions, as clearly instructed when creating an Issue.

**Comment by abonawas** [comment]
Yes, that is what I'm asserting. And I did not ignore the read.me, I provided an example which shows clearly that the residuals on the standard plot method are inverted.
Residuals are defined as observed data - fit, not the other way around.

**Comment by newville** [comment]
No, really the issue is closed.  As you were clearly instructed, begin with a discussion on the mailing list.  Any sympathy I may have had for your view is completely lost by your inability to follow clear instructions.
